hyperhtml.js

```diff
@@ -60,129 +47,105 @@ var hyperHTML = (function () {'use strict';
-              (i + 1 === length || childNodes[i + 1].nodeType === ELEMENT_NODE)
-            ) {
-              actions.push(remapping ?
-                {a:'vc', n:child} : setVirtualContent(child));
-            } else {
-              if (remapping) {
-                actions.push({a:'text', n:child});
-              } else {
-                text = node.ownerDocument.createTextNode('');
-                actions.push(setTextContent(text));
-                node.replaceChild(text, child);
-              }
-            }
-          }
-          break;
-        case TEXT_NODE:
-          // TODO: once SHOULD_USE_ATTRIBUTE contains more attributes
-          //       it's probably a good idea not to use it in here.
-          if (SHOULD_USE_ATTRIBUTE.test(node.nodeName) && child.textContent === uidc) {
-            actions.push(remapping ? {a:'text', n:node} : setTextContent(node));
-          }
+  // accordingly with the content type
+  // it replace the content of a node
+  // with the give child
+  function populateNode(parent, child) {
+    switch (child.nodeType) {
+      case ELEMENT_NODE:
+        var childNodes = parent.childNodes;
+        if (childNodes[0] === child) {
+          removeNodeList(childNodes, 1);
           break;
-      }
+        }
+        resetAndPopulate(parent, child);
+        break;
+      case DOCUMENT_FRAGMENT_NODE:
+        if (-1 < indexOfDifferences(parent.childNodes, child.childNodes)) {
+          resetAndPopulate(parent, child);
+        }
+        break;
+      case TEXT_NODE:
+        parent.textContent = child.textContent;
+        break;
     }
   }
 
+  // remove a list of nodes from startIndex to list.length
+  function removeNodeList(list, startIndex) {
+    var length = list.length, child;
+    while (startIndex < length--) {
+      child = list[length];
+      child.parentNode.removeChild(child);
+    }
+  }
+
+  // erase a node content and populate it
+  function resetAndPopulate(parent, child) {
+    parent.textContent = '';
+    parent.appendChild(child);
+  }
+
+  // append childNodes to a node from a specific index
+  function updateViaArray(node, childNodes, i) {
+    var fragment = emptyFragment(node);
+    if (0 < i) {
+      removeNodeList(node.childNodes, i);
+      appendNodes(fragment, childNodes.slice(i));
+      node.appendChild(fragment);
+    } else {
+      appendNodes(fragment, childNodes);
+      resetAndPopulate(node, fragment);
+    }
+  }
 
-  // -------------------------
-  // DOM manipulating
-  // -------------------------
+  // ---------------------------------------------
+  // hyperHTML Operations
+  // ---------------------------------------------
 
-  // update regular bound nodes
-  //
-  // var render = hyperHTML.bind(node);
-  // function update() {
-  //  render`template`;
-  // }
+  // `<div>${'any'}</div>`
   function setAnyContent(node) {
     return function any(value) {
-      switch (typeof value) {
+      var type = typeof value;
```

Comment:
you are comparing with previous function but it's been fully rewritten.
```js
  // `<div>${'any'}</div>`
  function setAnyContent(node) {
    return function any(value) {
      var type = typeof value;
      switch (type) {
        case 'string':
        case 'number':
        case 'boolean':
          node.innerHTML = value;
          break;
        default:
          if (isArray(value)) {
            var length = value.length;
            if (length === 1) {
              any(value[0]);
            } else {
              if (0 < length) type = typeof value[0];
              if (type === 'string') {
                any(value.join(''));
              } else {
                var i = indexOfDifferences(node.childNodes, value);
                if (-1 < i) updateViaArray(node, value, i);
              }
            }
          } else {
            populateNode(node, value);
          }
          break;
      }
    };
  }
```
I don't see any issue there